controllers/elasticsearch/elasticsearch_controller.go

```diff
@@ -0,0 +1,37 @@
+package elasticsearch
+
+import (
+	"context"
+
+	esv1 "github.com/openshift/elasticsearch-operator/apis/logging/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/jaegertracing/jaeger-operator/pkg/controller/elasticsearch"
+)
+
+// ElasticsearchReconciler reconciles a Deployment object
+type ElasticsearchReconciler struct {
+	reconcilier *elasticsearch.ReconcileElasticsearch
+}
+
+// NewElasticsearchReconciler creates a new deployment reconciler controller
+func NewElasticsearchReconciler(client client.Client, clientReader client.Reader) *ElasticsearchReconciler {
```

Comment:
To be honest I am not sure whether we should use controller or webook to catch changes on the ES CR. I didn't find any good docs when to use webhook over controller.

Reply:
I think for this case it is OK to use a controller, I haven't found a reference for when to use webhooks or controllers, but I tend to thing that webhooks are for validation or mutating the resource, which is not the case here.  Also this allow us to reenqueue the reconciliation in case of failure.

Reply:
I think the controller is fine here, the webhooks should be more used for validation or mutating. 

I had to make one more important change. The controller can be installed only if `Elasticsearch` kind is installed in the cluster (OpenShift Elasticserach operator).

Reply:
It seems this is already covered in docs

https://docs.openshift.com/container-platform/4.9/distr_tracing/distr_tracing_install/distr-tracing-installing.html#distr-tracing-jaeger-operator-install_install-distributed-tracing

> If you require persistent storage, you must also install the OpenShift Elasticsearch Operator before installing the Red Hat OpenShift distributed tracing platform Operator.

cc) @JStickler